File: Python3/Min Cost Climbing Stairs.py
# 48ms, 65.90%; 13.3MB, 7.01%
class Solution:
    def minCostClimbingStairs(self, cost: list) -> int:
        cost.append(0)
        dp = [cost[0], cost[1]] + [0] * (len(cost) - 2)
        for i in range(2, len(cost)):
            dp[i] = cost[i] + min(dp[i - 2], dp[i - 1])
        return dp[-1]

File: Cpp/Min Cost Climbing Stairs.cpp
// 8ms, 99.84%; 8.9MB, 98.94%
static int x = []() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution
{
public:
	int minCostClimbingStairs(vector<int> &cost)
	{
		cost.push_back(0);
		int size = cost.size();
		vector<int> dp{cost[0], cost[1]};
		dp.resize(size);
		for(int i = 2; i < size; ++i)
			dp[i] = cost[i] + min(dp[i - 2], dp[i - 1]);
		return dp.back();
	}
};
